fix(DB/Zone):improved Western Plaguelands

Knindzagxg opened the pull request:
Western Plaguelands are now fully updated, since some creatures in zone was missing SmartAI.

Working on every zone at the moment, and will post updates once i'm done with each.

To test it:
Apply SQL
Restart your server

You can manually spawn NPC by ID to check SmartAI or you can go to zone and check like that as well.

.npc add ID
or
.tele to zone and run / test around.

pklloveyou commented:
Tested

Si1ker commented:
**TEST: 3 failed** 🐱 @FrancescoBorzi @Knindzagxg 

Spectral Attendant - On Aggro - Cast \'11443\''),
![image](https://user-images.githubusercontent.com/61223313/94987729-b09dea00-0525-11eb-9dc2-4c886db149ac.png)

Spectral Attendant - In Combat - Cast \'13860\''),
![image](https://user-images.githubusercontent.com/61223313/94987730-b5fb3480-0525-11eb-9a76-c4dd54759e87.png)

Spectral Attendant - Between 30-60% Health - Cast \'11981\' (No Repeat)');
![image](https://user-images.githubusercontent.com/61223313/94987733-bb587f00-0525-11eb-87f5-70415fbbaf8b.png)

============================
'Fetid Zombie - In Combat - Cast \'8714\'');
![image](https://user-images.githubusercontent.com/61223313/94987768-15f1db00-0526-11eb-9d09-0cda4c6c9613.png)


============================
Haunting Vision - On Aggro - Cast \'11443\''),
![image](https://user-images.githubusercontent.com/61223313/94987840-81d44380-0526-11eb-90f4-c98a0fc1a274.png)

Haunting Vision - In Combat - Cast \'13860\''),
![image](https://user-images.githubusercontent.com/61223313/94987843-8698f780-0526-11eb-948c-761b589041da.png)

'Haunting Vision - Between 30-60% Health - Cast \'11981\' (No Repeat)');
![image](https://user-images.githubusercontent.com/61223313/94987856-90225f80-0526-11eb-8932-36090ab558fa.png)


============================
Decaying Horror - On Aggro - Cast \'32065\'');
![image](https://user-images.githubusercontent.com/61223313/94987902-df689000-0526-11eb-920e-7017919a8462.png)


============================
'Plague Lurker - Between 20-80% Health - Cast \'3436\' (No Repeat)');
![image](https://user-images.githubusercontent.com/61223313/94987932-0d4dd480-0527-11eb-8e6a-283718f74fdf.png)


Huntsman Radley - On Aggro - Summon Creature \'Bloodshot\''),
![image](https://user-images.githubusercontent.com/61223313/94987978-52720680-0527-11eb-8994-b0665e65da75.png)


'Huntsman Radley - In Combat - Cast \'11976\''),
![image](https://user-images.githubusercontent.com/61223313/94988008-6e75a800-0527-11eb-8207-7b588383379b.png)


'Huntsman Radley - Between 20-80% Health - Cast \'14443\' (No Repeat)');
### [FAILED]
 I can't get this one to work. reading the spell description it says:

![image](https://user-images.githubusercontent.com/61223313/94988069-cad8c780-0527-11eb-8355-52298d087b6b.png)

I have tried to run away in order to allow the distance but it just does not happen  : /


============================
'Bloodshot - Between 20-80% Health - Cast \'13692\' (No Repeat)'),
![image](https://user-images.githubusercontent.com/61223313/94988188-b34e0e80-0528-11eb-830e-ee69e5548206.png)

'Bloodshot - On Evade - Despawn In 5000 ms');
![image](https://user-images.githubusercontent.com/61223313/94988193-bb0db300-0528-11eb-842d-b6e8f91acdc2.png)

( It's fading away .... )


============================
Scarlet Cavalier - In Combat - Cast \'16856\''),
![image](https://user-images.githubusercontent.com/61223313/94988250-15a70f00-0529-11eb-9cf4-d8bea3b0d6fb.png)



'Scarlet Cavalier - Between 40-80% Health - Cast \'15284\' (No Repeat)'),
![image](https://user-images.githubusercontent.com/61223313/94988441-6bc88200-052a-11eb-87ca-deecc277cd4f.png)


Scarlet Cavalier - Between 10-30% Health - Cast \'6253\' (No Repeat)');
![image](https://user-images.githubusercontent.com/61223313/94988439-69662800-052a-11eb-931d-5be3d2e69cb1.png)


============================
'Scarlet Priest - On Respawn - Cast \'13864\''),
![image](https://user-images.githubusercontent.com/61223313/94988494-d083dc80-052a-11eb-9f63-3bd63fedfac6.png)
### [FAILED]
When the NPC spawns, It will cast 13864 at the first only. If you leave combat with the NPC, the aura will be gone, if you engaged the NPC once again, it will not use it, this is not correct. Take for example the NPC with ID 1889, this NPC uses a shield (12544) that will always use, at reset it recast the spell so the npc is always looking to have it up/active at all time (meaning in combat and out of combat), same way it should be working here. OR use it at agro instead of respawn, your choice.



 'Scarlet Priest - In Combat - Cast \'15498\''),
![image](https://user-images.githubusercontent.com/61223313/94988512-f9a46d00-052a-11eb-922a-059b5a36b854.png)


Scarlet Priest - Between 60-90% Health - Cast \'8362\' (No Repeat)'),
![image](https://user-images.githubusercontent.com/61223313/94988515-01641180-052b-11eb-8112-a31882764993.png)


'Scarlet Priest - Between 5-25% Health - Cast \'12039\' (No Repeat)');
![image](https://user-images.githubusercontent.com/61223313/94988544-48520700-052b-11eb-95d6-9dd95b0cb6c1.png)


============================
'Cavalier Durgen - On Aggro - Cast \'13008\''),
![image](https://user-images.githubusercontent.com/61223313/94988568-7fc0b380-052b-11eb-8760-e41476185333.png)


 'Cavalier Durgen - In Combat - Cast \'13953\''),
![image](https://user-images.githubusercontent.com/61223313/94988575-894a1b80-052b-11eb-80a1-0bb1e8d4f18e.png)


 'Cavalier Durgen - Between 10-30% Health - Cast \'13005\' (No Repeat)');
![image](https://user-images.githubusercontent.com/61223313/94988608-b0085200-052b-11eb-8170-51c10aa0f7ab.png)


============================
'Scarlet Lumberjack - In Combat - Cast \'11977\''),
![image](https://user-images.githubusercontent.com/61223313/94988633-dcbc6980-052b-11eb-8c94-24611f8a49f4.png)


'Scarlet Lumberjack - Between 25-40% Health - Cast \'15496\' (No Repeat)');
![image](https://user-images.githubusercontent.com/61223313/94988636-e940c200-052b-11eb-9a2d-058e97173841.png)


============================
 'Skeletal Terror - On Aggro - Cast \'17650\''),
![image](https://user-images.githubusercontent.com/61223313/94988680-548a9400-052c-11eb-8e80-d80917625270.png)


 'Skeletal Terror - Between 5-30% Health - Cast \'12542\' (No Repeat)');
![image](https://user-images.githubusercontent.com/61223313/94988690-61a78300-052c-11eb-9cfa-d6bbd07b444d.png)

============================
 'Scarlet Medic - On Respawn - Cast \'17175\''),
![image](https://user-images.githubusercontent.com/61223313/94988722-9287b800-052c-11eb-9485-67ccf8a51f67.png)
### [FAILED]
When the NPC spawns, It will cast 17175at the first only. If you leave combat with the NPC, the aura will be gone, if you engaged the NPC once again, it will not use it, this is not correct. Take for example the NPC with ID 1889, this NPC uses a shield (12544) that will always use, at reset it recast the spell so the npc is always looking to have it up/active at all time (meaning in combat and out of combat), same way it should be working here. OR use it at agro instead of respawn, your choice.


 'Scarlet Medic - Between 5-95% Health - Cast \'11640\' (No Repeat)'),
![image](https://user-images.githubusercontent.com/61223313/94988740-b64afe00-052c-11eb-8bda-c32a4bfb96fe.png)


 'Scarlet Medic - Between 5-20% Health - Cast \'17137\' (No Repeat)');
![image](https://user-images.githubusercontent.com/61223313/94988751-d5499000-052c-11eb-90a1-847c9791b875.png)


============================
'Jabbering Ghoul - On Aggro - Cast \'15716\''),
![image](https://user-images.githubusercontent.com/61223313/94988805-2d809200-052d-11eb-8cd6-7c1a4a87e0a1.png)


'Jabbering Ghoul - In Combat - Cast \'12097\'');
![image](https://user-images.githubusercontent.com/61223313/94988829-56a12280-052d-11eb-9851-6159ed4143a0.png)


============================
 'Venom Mist Lurker - In Combat - Cast \'3583\' (No Repeat)');
![image](https://user-images.githubusercontent.com/61223313/94988853-7df7ef80-052d-11eb-8739-b2c4f0968a70.png)

Knindzagxg commented:
Jesus those tests xD

Updated Priest and Medic as suggested xP Weird it's droping btw, never saw it before.
Updated Auras as suggested as well xP

Gonna look  a bit more into Huntsman, cause he did cast few times on me.

Knindzagxg commented:
Sorted Huntsman.

For some reason, Hostile_random not working at all.
Tried with multiple chars but no luck at all.
At least its good to know, so we can avoid until its fixed.

Recheck if u can again @Si1ker xD

Si1ker commented on data/sql/updates/pending_db_world/rev_1601314686326215700.sql:
```diff
@@ -40,7 +40,7 @@ DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 11613);
 (11613, 0, 0, 0, 4, 0, 100, 0, 0, 0, 0, 0, 12, 11614, 1, 30000, 1, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Huntsman Radley - On Aggro - Summon Creature \'Bloodshot\''),
 (11613, 0, 1, 0, 0, 0, 100, 0, 2700, 3900, 9700, 12400, 11, 11976, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Huntsman Radley - In Combat - Cast \'11976\''),
-(11613, 0, 2, 0, 2, 0, 100, 1, 20, 80, 0, 0, 11, 14443, 0, 0, 0, 0, 0, 5, 0, 0, 0, 0, 0, 0, 0, 'Huntsman Radley - Between 20-80% Health - Cast \'14443\' (No Repeat)');
+(11613, 0, 2, 0, 2, 0, 100, 1, 20, 80, 0, 0, 11, 14443, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Huntsman Radley - Between 20-80% Health - Cast \'14443\' (No Repeat)');
```
@Knindzagxg This one is still not working...  :,c

Just melees and her 11976..
![image](https://user-images.githubusercontent.com/61223313/94999867-a279b900-0579-11eb-9b9d-d55ff10142da.png)

Knindzagxg replied:
It's working, just make sure to make small macro and keep doing damage while away more then 5 yards xP

Si1ker replied:
@Knindzagxg Believe me, is not working..  :'c

![image](https://user-images.githubusercontent.com/61223313/95020697-e2987480-0629-11eb-9817-bbad6e309b31.png)

Si1ker commented on data/sql/updates/pending_db_world/rev_1601314686326215700.sql:
```diff
@@ -40,7 +40,7 @@ DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 11613);
 INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `t ...
 (11613, 0, 0, 0, 4, 0, 100, 0, 0, 0, 0, 0, 12, 11614, 1, 30000, 1, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Huntsman Radley - On Aggro - Summon Creature \'Bloodshot\''),
```
@Knindzagxg  I noticed something interesting this time about this...
She already has one out at normal spawn... so when you get in combat with her, she will execute this and spawn the second one, is still ok that she has 2 of this minion at combat then (the original + the summon)?

The original and permanent one is:
NPC ID: 11614
GUID: 52640
Coords: X= 2328.97  Y= -2070.82  Z= 168.62
![image](https://user-images.githubusercontent.com/61223313/94999961-3f3c5680-057a-11eb-92ff-0a3608e939b3.png)

Knindzagxg replied:
Looks like Blizz decided to not use spell for spawn and have it up always.
Sorted it as well.

Si1ker commented:
Cavalier Durgen 
Scarlet Priest
Scarlet Medic

These are just fine now 🐱

Si1ker commented:
Minion summon resolved, now just the multi-shot thing is left. 🐱

Knindzagxg commented:
It's sorted now xP
Just unlucky procs with yard and hp % xP

Si1ker reviewed (approved):
Let's gooo ! 🐱
